src/charm.py

```diff
@@ -41,83 +65,51 @@ def __init__(self, *args):
 
     def _update_prometheus_jobs(self, _):
         """Setup Prometheus scrape configuration for external targets."""
-        self.unit.status = ActiveStatus()
-        if not self.unit.is_leader():
-            return
-
-        jobs = self._scrape_jobs()
-        if not jobs:
-            return
-
-        for relation in self.model.relations[self._prometheus_relation]:
-            relation.data[self.app]["scrape_jobs"] = json.dumps(jobs)
+        if self.unit.is_leader():
+            if jobs := self._scrape_jobs():
+                for relation in self.model.relations[self._prometheus_relation]:
+                    relation.data[self.app]["scrape_jobs"] = json.dumps(jobs)
 
-    def _scrape_jobs(self):
-        targets = self._targets()
+                self.unit.status = ActiveStatus()
 
-        jobs = (
-            [
+    def _scrape_jobs(self) -> list:
+        if targets := self._targets():
+            labels = self._labels()
+            return [
                 {
                     "job_name": self._job_name(),
                     "metrics_path": self.model.config["metrics-path"],
                     "static_configs": [
                         {
                             "targets": targets,
+                            "labels": labels,
                         }
                     ],
                 }
             ]
-            if targets
-            else []
-        )
-
-        labels = self._labels()
-        if jobs and labels:
-            jobs[0]["static_configs"][0]["labels"] = labels
 
-        return jobs
+        return []
 
-    def _targets(self):
-        if not (targets := self.model.config.get("targets", "")):
+    def _targets(self) -> list:
+        if not (config_targets := self.model.config.get("targets", "")):
```

Comment:
If all you were trying to do here is change a variable name to your preference then perhaps a better variable name would have been `scrape_targets` because that is what they are. They certainly are not `config_targets`.

Reply:
I think the config_ prefix is because it is taken from the config and is not yet a "confirmed" scrape target

Reply:
Hmm. I don't see where the question of "confirmation" arises. The charm has no procedure to "confirm" the targets. It does validate them. So if you want to be more verbose `unverified_scrape_targets`.

Reply:
Renamed.